src/google/protobuf/compiler/command_line_interface.cc

```diff
@@ -2228,12 +2228,18 @@ bool CommandLineInterface::WriteDescriptorSet(
   }
 
   io::FileOutputStream out(fd);
-  if (!file_set.SerializeToZeroCopyStream(&out)) {
+
+  io::CodedOutputStream coded_out(&out);
+  // Determinism is useful here because build outputs are sometimes checked
+  // into version control.
+  coded_out.SetSerializationDeterministic(true);
+  if (!file_set.SerializeToCodedStream(&coded_out)) {
     std::cerr << descriptor_set_out_name_ << ": " << strerror(out.GetErrno())
               << std::endl;
     out.Close();
     return false;
   }
+  coded_out.Trim();
```

Comment:
We usually just create the CodedOutputStream inside a block with curly braces: that way the destructor will take care of any cleanup and this Trim() call won't be needed.

Reply:
Done